hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -222,32 +222,41 @@ public class RocksDBCheckpointDiffer implements AutoCloseable,
         OZONE_OM_SNAPSHOT_COMPACTION_DAG_MAX_TIME_ALLOWED_DEFAULT,
         TimeUnit.MILLISECONDS);
 
-    long pruneCompactionDagDaemonRunIntervalInMs =
+    this.pruneCompactionDagDaemonRunIntervalInMs =
         configuration.getTimeDuration(
             OZONE_OM_SNAPSHOT_COMPACTION_DAG_PRUNE_DAEMON_RUN_INTERVAL,
             OZONE_OM_SNAPSHOT_PRUNE_COMPACTION_DAG_DAEMON_RUN_INTERVAL_DEFAULT,
             TimeUnit.MILLISECONDS);
 
     if (pruneCompactionDagDaemonRunIntervalInMs > 0) {
-      this.executor = Executors.newSingleThreadScheduledExecutor();
-      this.executor.scheduleWithFixedDelay(
-          this::pruneOlderSnapshotsWithCompactionHistory,
-          pruneCompactionDagDaemonRunIntervalInMs,
-          pruneCompactionDagDaemonRunIntervalInMs,
-          TimeUnit.MILLISECONDS);
-
-      this.executor.scheduleWithFixedDelay(
-          this::pruneSstFiles,
-          pruneCompactionDagDaemonRunIntervalInMs,
-          pruneCompactionDagDaemonRunIntervalInMs,
-          TimeUnit.MILLISECONDS
-      );
+      this.scheduler = new Scheduler(dagPruningServiceName,
+          true, 1);
+      submitPruneJobs();
```

Comment:
it would be better to call the start method here.

Reply:
currently the closed flag is also true. Function doesn't change this flag.

Reply:
Yes, I just noticed that.

Reply:
Made the changed as suggested.